### PR TITLE
chore: add extra log properties

### PR DESCRIPTION
**Pull request by austinwoon**
Adds extra properties to logger context for better observability

- `id`: for better traces across log invocations
- `clientIp`: to track ip of client - helpful during instances where you need to block ip addresses in case of bots

### CODE REVIEW

File: src/utils/getClientIp.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { type NextApiRequest } from 'next'
+
+export default function getIP(request: Request | NextApiRequest) {
```
Comment (austinwoon) [inline src/utils/getClientIp.ts:3]: copied from nextjs example repo

**Review by karrui — APPROVED** [review]
Awesome. Was about to implement my own request id functionality but you beat me to it!

**Comment by karrui** [comment]
Tested failed, probably from the context change. Do you want to make the changes?

**Comment by austinwoon** [comment]
> Tested failed, probably from the context change. Do you want to make the changes?

Create a new util to mock the request context, see if you like that pattern, otherwise feel free to change it!

**Pull request merged**